src/hivpy/hiv_testing.py

```diff
@@ -50,81 +55,125 @@ def __init__(self, **kwargs):
+        self.test_mark_hiv_symptomatic(pop)
+        self.test_mark_non_hiv_symptomatic(pop)
+        self.test_mark_vmmc(pop, time_step)
+        self.test_mark_anc(pop, time_step)
+        self.test_mark_general_pop(pop)
 
-            # update testing probabilities
-            self.rate_first_test = self.init_rate_first_test + (min(pop.date.year, self.date_test_rate_plateau)
-                                                                - self.date_start_testing) \
-                                                                * self.an_lin_incr_test
-            self.rate_rep_test = (min(pop.date.year, self.date_test_rate_plateau)
-                                  - self.date_start_testing) * self.an_lin_incr_test
+        # apply testing to marked population
+        marked_population = pop.get_sub_pop([(col.TEST_MARK, op.eq, True)])
+        self.apply_test_outcomes_to_sub_pop(pop, marked_population)
 
-            # get population ready for testing
-            testing_population = pop.get_sub_pop([(col.HARD_REACH, op.eq, False),
-                                                  (col.AGE, op.ge, 15),
-                                                  (col.HIV_STATUS, op.eq, False),
-                                                  [(col.LAST_TEST_DATE, op.le, pop.date -
-                                                    timedelta(days=self.days_to_wait[self.eff_max_freq_testing])),
-                                                   (col.LAST_TEST_DATE, op.eq, None)]
-                                                  ])
-
-            # first time testers
-            untested_population = pop.apply_bool_mask(~pop.get_variable(col.EVER_TESTED, testing_population),
-                                                      testing_population)
-            # repeat testers
-            prev_tested_population = pop.apply_bool_mask(pop.get_variable(col.EVER_TESTED, testing_population),
-                                                         testing_population)
-
-            if len(untested_population) > 0:
-                # test first time testers
-                tested = pop.transform_group([col.EVER_TESTED, col.NP_LAST_TEST, col.NSTP_LAST_TEST],
-                                             self.calc_testing_outcomes,
-                                             sub_pop=untested_population)
+    def test_mark_hiv_symptomatic(self, pop):
+        """
+        Mark HIV symptomatic individuals to undergo testing this time step.
+        """
+        # testing occurs after a certain year
+        if (pop.date.year >= self.date_start_testing):
+
+            # update symptomatic test probabilities
+            if pop.date.year <= 2015:
+                self.prob_test_who4 = min(0.9, self.prob_test_who4 * self.incr_test_rate_sympt)
+                self.prob_test_tb = min(0.8, self.prob_test_tb * self.incr_test_rate_sympt)
+                self.prob_test_non_tb_who3 = min(0.7, self.prob_test_non_tb_who3 * self.incr_test_rate_sympt)
```

Comment:
This has a potential issue because it relies on the cumulative effects of `inc_test_rate_sympt` each timestep, and presume that a simulation _must_ start before `date_start_testing`. I expect that we would always run from before this date, but it is important to note that this code will not work if the run starts after that date, and at the very least that needs to be clearly documented somewhere.